Sources/MongoSwift/ConnectionPool.swift

```diff
@@ -1,12 +1,24 @@
 import mongoc
 
 /// A connection to the database.
-internal struct Connection {
+internal class Connection {
     /// Pointer to the underlying `mongoc_client_t`.
     internal let clientHandle: OpaquePointer
+    /// The pool this connection belongs to.
+    private let pool: ConnectionPool
 
-    internal init(_ clientHandle: OpaquePointer) {
+    internal init(clientHandle: OpaquePointer, pool: ConnectionPool) {
         self.clientHandle = clientHandle
+        self.pool = pool
+    }
+
+    deinit {
+        switch self.pool.state {
+        case let .open(pool):
+            mongoc_client_pool_push(pool, self.clientHandle)
+        case .closed:
+            assertionFailure("ConnectionPool was already closed")
```

Comment:
this was previously a `fatalError` but I think it is maybe better to not crash in release mode because of this... similar to our choice in the design use use `assert`s to make sure `close` has been called on e.g. `MongoClient`. what do you think?